engine/src/main/scala/cromwell/engine/workflow/InitializeWorkflowBackendsActor.scala

```diff
@@ -0,0 +1,86 @@
+package cromwell.engine.workflow
+
+import akka.actor.{Props, ActorRef, LoggingFSM}
+import cromwell.backend.WorkflowBackendActor.{InitializationFailed, InitializationSuccess, Initialize}
+import cromwell.engine.workflow.InitializeWorkflowBackendsActor._
+
+object InitializeWorkflowBackendsActor {
+
+  /**
+    * States
+    */
+  sealed trait InitializeWorkflowBackendsActorState
+  case object Unstarted extends InitializeWorkflowBackendsActorState
+  case object InitializingBackends extends InitializeWorkflowBackendsActorState
+  case object NoraDunn extends InitializeWorkflowBackendsActorState
+
+  /**
+    * State data
+    */
+  case class InitializeWorkflowBackendsActorData(successes: Seq[ActorRef], failures: Map[ActorRef, Throwable])
+
+  /**
+    * Commands
+    */
+  sealed trait InitializeWorkflowBackendsActorCommand
+  case object StartInitializations extends InitializeWorkflowBackendsActorCommand
+
+  /**
+    * Responses
+    */
+  sealed trait InitializeWorkflowBackendsActorResponse
+  case object BackendInitializationsSucceeded extends InitializeWorkflowBackendsActorResponse
+  case class BackendInitializationsFailed(reasons: Map[ActorRef, Throwable]) extends InitializeWorkflowBackendsActorResponse
+
+  def props(workflowTag: String, backendsToInitialize: Seq[ActorRef]): Props = Props(new InitializeWorkflowBackendsActor(workflowTag, backendsToInitialize))
+
+}
+
+case class InitializeWorkflowBackendsActor(workflowTag: String, backendsToInitialize: Seq[ActorRef]) extends LoggingFSM[InitializeWorkflowBackendsActorState, InitializeWorkflowBackendsActorData] {
+
+  startWith(Unstarted, InitializeWorkflowBackendsActorData(Seq.empty, Map.empty))
+
+  when(Unstarted) {
+    case Event(StartInitializations, _) =>
+      backendsToInitialize foreach { _ ! Initialize }
+      goto(InitializingBackends)
+  }
+
+  when(InitializingBackends) {
+    case Event(InitializationSuccess, stateData) =>
+      val newData = InitializeWorkflowBackendsActorData(stateData.successes ++ Seq(sender), stateData.failures)
+      checkForDoneAndTransition(newData)
+    case Event(InitializationFailed(reason), stateData) =>
+      val newData = InitializeWorkflowBackendsActorData(stateData.successes, stateData.failures ++ Map(sender -> reason))
+      checkForDoneAndTransition(newData)
+  }
+
+  private def checkForDoneAndTransition(newData: InitializeWorkflowBackendsActorData): State = {
+    if(checkForDone(newData)) {
+      if(newData.failures.isEmpty) {
```

Comment:
You could kinda collapse this a bit by assigning `BackendInitializationsSucceeded` and `BackendInit..Failed` to a val based on the `newData.failures` and then send that val to the parent